Feature/#19 unit tesztek irasa be n

czetta opened the pull request:
output file needed for upload tests

czetta committed:
Test package renamed

czetta committed:
Test package renamed

czetta committed:
Test package renamed

czetta committed:
Try to write useful tests

czetta committed:
Try to write useful tests

czetta committed:
Try to write useful tests

czetta committed:
Merge branch 'feature/#19_unit_tesztek_irasa_be-n' of
https://github.com/lapincsmatyas/KeneEgyCsapatnev into
feature/#19_unit_tesztek_irasa_be-n

# Conflicts:
#	caff-backend/src/main/java/com/example/cb/service/CaffService.java
#	caff-backend/src/test/java/com/example/cb/service/CaffServiceTests.java
#	caff-backend/src/test/java/com/example/cb/service/CommentServiceTests.java

czetta committed:
CaffService tests written, upload tests need to be overviewed with
output file present

czetta committed:
CaffService tests written, upload tests need to be overviewed with
output file present

czetta committed:
test files added

czetta committed:
test files added

lapincsmatyas commented on caff-backend/src/main/java/com/example/cb/repository/CaffRepository.java:
```diff
@@ -7,4 +7,5 @@
 
 @Repository
 public interface CaffRepository extends JpaRepository<Caff, Long> {
```
Ha a repository a CrudRepositoryból öröklődik, akkor tartalmazza a findById függvényt alapból :) 
Egyelőre jó lesz, majd átírom, mert valószítű akkor több helyen is JpaRepository az ős, csak hogy lásd, hogy ilyen is van :)

lapincsmatyas commented on caff-backend/src/main/java/com/example/cb/repository/CaffRepository.java:
```diff
@@ -7,4 +7,5 @@
 
 @Repository
 public interface CaffRepository extends JpaRepository<Caff, Long> {
+	public Caff findById(long id);
```
A Repository-k mindig optinallal térjenek vissza :) Ez a "helyes" működésük. Ez egy egyszerű csomgóló objektum, amiről a kezelő kód el tudja dönteni, hogy van-e benne érték vagy nincs. Ezután tud mindenféle logikát rápakolni.

lapincsmatyas commented on caff-backend/src/main/java/com/example/cb/service/CaffService.java:
```diff
@@ -117,8 +117,8 @@ public Caff execCmd(String[] command) throws ParseException, InterruptedExceptio
-    public Caff getCAFFById(Long id) {
-        return caffRepository.findById(id).get();
+    public Caff getCAFFById(long id) {
+        return caffRepository.findById(id);
```
Maradjon a .get() és legyen Optional a repository findById visszatérése

lapincsmatyas commented on caff-backend/src/test/java/com/example/cb/service/CaffServiceTests.java:
```diff
@@ -0,0 +1,161 @@
+	void testUploadCaff() throws ParseException, IOException {
+		CaffFile caffFile = new CaffFile();
+		String path = "src/test/java/com/example/cb/service/test_caff/1.caff";
+		File file = new File(path);
```
annyira nem fontos változás, de később inkább használd a NIO packaget. 
Paths.get() pl. visszaadja az elérését a fájlnak, a Files statikus metódusai pedig megnyitják, létrehozzák, stb. azt

czetta committed:
Test package renamed

Conflicts:
	caff-backend/src/test/java/com/example/cb/CaffBackendApplicationTests.java

czetta committed:
Merge branch 'feature/#19_unit_tesztek_irasa_be-n' of https://github.com/lapincsmatyas/KeneEgyCsapatnev into feature/#19_unit_tesztek_irasa_be-n

czetta committed:
Tests updated, compatible with md5.

czetta committed:
CaffService tests written, upload tests not tested

czetta commented:
Az upload-hoz tartozó teszteket output.exe híján nem tudtam kipróbálni, hogy működnek-e. Az uploadinvalidcaff tesztet kikommenteztem, az lehet hogy nem jó, a másiknak elvileg működnie kell, ha van parser, azt bennehagytam.

czetta committed:
Merge branch 'feature/upload_caff_without_file' into feature/#19_unit_tesztek_irasa_be-n
